### DISCUSSION TITLE
Which versions of discord.js will EOL this month?

### DISCUSSION THREAD

**Discussion by jeremybradbury** [discordjs/discord.js#7815] [Q&A] [answered]
After searching docs, I know v13 uses API v9 but I am unsure when/if v12 made the switch to v8 and if we can just upgrade 12 a bit or need to switch to 13 very quickly.

Perhaps I've missed it in the changelogs or something, but it'd be sweet to see a short list of releases paired to Discord API version changes, somewhere in the current docs. 

If not perhaps a long time maintainer could leave (or start) a list like that here?

**Comment by SuperchupuDev** [comment]
discord.js v12 hasn't been supported for quite a while, and uses API v7, however Discord is delaying the API v7 decommission date to 2023 (see discord/discord-api-docs#4657). Even then you should be updating to v13 soon, since v12 can break at any time and is already breaking with new upcoming features (text in voice)

**Comment by jeremybradbury** [comment]
@SuperchupuDev TYVM!

>  since v12 can break at any time and is already breaking with new upcoming features (text in voice)

yes I know we've had an extensive custom bot that edits an embedded post with the latest details among many other things. it's been hard to justify any changes, but we should have that server on node 16 too. when stuff just works, it often ends up in the "happily out of maintenance" category.

> however Discord is delaying the API v7 decommission date to 2023 (see https://github.com/discord/discord-api-docs/discussions/4657)

this truly made my day, i know we have to have this done, just wasn't excited about having it done this month (if it was v6 or something).

Just for clarity did sdk v12 launch on api v7 or was there a minor/patch it changed from 6 to 7? 

I may have to at least test a minor update this month if that's the case.

**Reply by SuperchupuDev** [reply, marked as answer]
[Apparently](https://github.com/discordjs/discord.js/blob/12.0.0/src/util/Constants.js#L82) discord.js v12 launched with API v7, but anyways Discord ended up moving the decommission date for API v6 too some days after they announced it for v7
- discord/discord-api-docs#4725

**Reply by jeremybradbury** [reply]
@SuperchupuDev your badassery knows no bounds. 

Sorry for all the stupid questions.

I hope this thread saves some others a little bit of time!
